src/black/__init__.py

```diff
@@ -602,6 +612,18 @@ def main(
     ctx.exit(report.return_code)
 
 
+def test_regex(
+    ctx: click.Context,
+    regex_name: str,
+    regex: Optional[str],
+) -> Optional[Pattern]:
```

Comment:
I don't know the `mypy` incantation to say this returns an optional if `regex` was an optional. Suggestions welcome 👍

Reply:
You'd have to do it with overloads, but not sure that's worth bothering with. Something like 

`@overload
def test_regex(regex: str) -> Pattern: ...
@overload
def test_regex(regex: None) -> None: ...`

Reply:
I suppose that makes sense.

> Looks like all that would gain us is removing an assert further down?

yup. Totally not worth 😄